src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const arrayOfAcrions = [];
+  let currentAction = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        let addAction = { ...currentAction };
+
+        addAction = Object.assign(addAction, action.extraData);
+
+        arrayOfAcrions.push(addAction);
+        currentAction = addAction;
```

Comment:
1. You create a copy of `currentAction`,
2. then assign some `extraData` to this copy,
3. then push copy to the actions' array
4. then assign copy back to `currentAction`

I think you don't have to take this additional copying step because you already made a copy of `state` object, and now can easily mutate `currentAction` object.

```suggestion
        Object.assign(currentAction, action.extraData);
```


The same is for `removeProperties` case